assets/js/base/utils/price.js

```diff
@@ -100,25 +106,29 @@ export const getCurrency = ( currencyData = {} ) => {
  * decimal complete with currency symbols using current store settings.
  *
  * @param {number|string} price Price in minor unit, e.g. cents.
- * @param {Object} currencyData Currency data object.
+ * @param {Object} [currencyData] Currency data object.
  */
-export const formatPrice = ( price, currencyData ) => {
+export const formatPriceAsString = ( price, currencyData = {} ) => {
 	if ( price === '' || price === undefined ) {
 		return '';
 	}
 
-	const priceInt = parseInt( price, 10 );
+	const priceInt = typeof price === 'string' ? parseInt( price, 10 ) : price;
 
 	if ( ! Number.isFinite( priceInt ) ) {
 		return '';
 	}
 
 	const currency = getCurrency( currencyData );
-	const formattedPrice = priceInt / 10 ** currency.minorUnit;
-	const formattedValue = currency.prefix + formattedPrice + currency.suffix;
+	const formattedPrice = renderToString(
+		<FormattedMonetaryAmount
+			currency={ currency }
+			displayType="text"
+			value={ priceInt }
+		/>
+	);
 
-	// This uses a textarea to magically decode HTML currency symbols.
-	const txt = document.createElement( 'textarea' );
-	txt.innerHTML = formattedValue;
-	return txt.value;
+	// Remove HTML.
+	const tagsRegExp = /<\/?[a-z][^>]*?>/gi;
+	return formattedPrice.replace( tagsRegExp, '' );
```

Comment:
Hmm...  interesting approach here. I originally did a double-take because I thought `renderToString` could only be used reliably in the server context but it looks like it [can be used in both environments now](https://reactjs.org/docs/react-dom-server.html#overview).

It'd be good to add to the tests some cases around removing html.